velox/expression/tests/utils/ArgumentTypeFuzzer.cpp

```diff
@@ -35,6 +38,91 @@ std::optional<TypeKind> baseNameToTypeKind(const std::string& typeName) {
   return tryMapNameToTypeKind(kindName);
 }
 
+namespace {
+
+bool isDecimalBaseName(const std::string& typeName) {
+  auto normalized = boost::algorithm::to_lower_copy(typeName);
+
+  return normalized == "decimal";
+}
+
+/// Returns true only if 'str' contains digits.
+bool isPositiveInteger(const std::string& str) {
+  return !str.empty() &&
+      std::find_if(str.begin(), str.end(), [](unsigned char c) {
+        return !std::isdigit(c);
+      }) == str.end();
+}
+
+int32_t rand(FuzzerGenerator& rng) {
+  return boost::random::uniform_int_distribution<int32_t>()(rng);
+}
+} // namespace
+
+void ArgumentTypeFuzzer::determineUnboundedIntegerVariables(
+    const exec::TypeSignature& type) {
+  if (!isDecimalBaseName(type.baseName())) {
+    return;
+  }
+
+  VELOX_CHECK_EQ(2, type.parameters().size())
+
+  const auto& precision = type.parameters()[0].baseName();
+  const auto& scale = type.parameters()[1].baseName();
+
+  // Bind 'name' variable, if not already bound, using 'constant' constraint
+  // ('name'='123'). Return bound value if 'name' is already bound or was
+  // successfully bound to a constant value. Return std::nullopt otherwise.
+  auto tryFixedBinding = [&](const auto& name) -> std::optional<int> {
+    auto it = variables().find(name);
+    if (it == variables().end()) {
+      return std::stoi(name);
+    }
+
+    if (integerBindings_.count(name) > 0) {
+      return integerBindings_[name];
+    }
+
+    if (isPositiveInteger(it->second.constraint())) {
+      const auto value = std::stoi(it->second.constraint());
```

Comment:
would it make sense to throw incase there are constraints other than positive constants? This would ensure someone does not unknowingly misuses it.
Should we also mention this limitation in the header file?

Reply:
@bikramSingh91 I'm not sure I understand. In most cases constraints are not constant, but rather expressions: s3 = max(s1, s2).

Reply:
Since this is not designed to fuzz arguments if a return type is provided and there are complex constraints, it is possible that a client can misuse it by providing a return type then calling `fuzzArgumentTypes` where L88 would return false and eventually return std::nullopt at L94 and `p` and `s` would be randomly assigned values. The client will never know that the complex constraint might not hold.

Reply:
fuzzArgumentTypes uses ReverseSignatureBinder to bind type and integer variables using return type. ReverseSignatureBinder::tryBind returns false if the signature has constraints.

```
    exec::ReverseSignatureBinder binder{signature_, returnType_};
    if (!binder.tryBind()) {
      return false;
    }


bool ReverseSignatureBinder::tryBind() {
  if (hasConstrainedIntegerVariable(signature_.returnType())) {
    return false;
  }
  return SignatureBinderBase::tryBind(signature_.returnType(), returnType_);
}
```